core/controllers/acl_decorators.py

```diff
@@ -2335,11 +2335,12 @@ def can_access_story_viewer_page(handler):
             if the user can access the given story viewer page.
     """
 
-    def test_can_access(self, story_id, **kwargs):
+    def test_can_access(self, story_id, *args, **kwargs):
```

Comment:
What's the reason for this change?

Reply:
Since the new function I added in RecommendationsHandler has both story_id and node_id as arguments, so *args was needed here.

Reply:
Then should node_id be part of the URL pattern?

Reply:
It's not needed in the URL. This decorator is also used when viewing a story viewer page in which case node_id is not needed. I added *args since, I am using this same decorator to validate `_record_completion_and_get_next_exp_id` as well, since that is used at the end of a story, where the node_id is needed to get the exploration ID of the next chapter.
This decorator itself does not use node_id.